### DISCUSSION TITLE
Installing vm-based Azure stack HCI lab on windows 11 pro laptop

### DISCUSSION THREAD

**Discussion by nm7s9** [Microsoft/WSLab#594] [General]
I'm trying to install the Azure stack HCI vm-based lab using instruction on

https://learn.microsoft.com/en-us/azure-stack/hci/deploy/tutorial-private-forest

I;m trying to install it on a 64GB ram laptop running Windows 11 Pro.
I realise that the instructions say to run it on Windows server 2022,2019 or 2016 but I did ask bing if it could be done on Windows 11 Pro and (copilot) said 

"Certainly! You can create a VM-based lab for Azure Stack HCI on a Windows 11 Pro PC. Here are the steps you’ll need to follow:
Prerequisites:
Admin privileges on a Hyper-V host server running Windows Server 2022, Windows Server 2019, or Windows Server 2016........"

I think that maybe copilot just told me what I wanted to hear, because....

When running the 2_CreateParentDisks.ps1 the first time it gets "stuck" repeatedly doing

                 Restarting DC
                 Configuration in Progress. Sleeping 10 seconds
                 Configuration in Progress. Sleeping 10 seconds
                 Configuration in Progress. Sleeping 10 seconds
                 Current DSC state: Failure, ResourncesNotInDesiredState: 20, ResourncesInDesiredState: 10.
                 Restarting DC
                 Configuration in Progress. Sleeping 10 seconds
                 Configuration in Progress. Sleeping 10 seconds

It's being doing this for 5 hours now.
If I log in to the DC vm that it has created and view the event log I see repeated liscence errors (slui.exe) which seems to indicate to me that it may not like the fact that I am running windows 11 Pro, but I could be wrong.

Any advise as to how I can install the vm-based lab on mhy laptop would be greatly appreciated.

**Comment by Karl-WE** [comment]
Can you confirm you have specified a Windows Server 2022 ISO in when launching this? 
see #588  #589

**Reply by nm7s9** [reply]
My laptop is Windows 11 Pro 23H2 and the two ISO I am using for the install are

PS C:\Lab\Isos> dir

    Directory: C:\Lab\Isos

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----        11/05/2024   8:01 PM     5652088832 17763.3650.221105-1748.rs5_release_svc_refresh_SERVER_EVAL_x64FRE_en-us.iso
-a----        11/05/2024   7:26 PM     3530342400 AzureStackHCI_20349.1607_en-us.iso

PS C:\Lab\Isos>

The eval server ISO Windows Server 2019

**Reply by Karl-WE** [reply]
2019 _should_ be fine, can you be bothered to try 2022 for the 2_CreateParentdisks.ps1, please. 
Could be that the DSC updates lately done in mslab broke something. please keep in mind that WS 2019 is no longer in mainstream support. so functional fixes getting rare and limited to security fixes.

**Reply by nm7s9** [reply]
Yep - I'll try tonight

**Reply by nm7s9** [reply]
Just tried Windows Server 2022 evaluation copy - but exactly the same issue

**Reply by nm7s9** [reply]
5/15/2024 10:01:24 PM  8198 Microsoft-Windows-Security-SPP                             License Activation (slui.exe) failed with the following error code:
                                                                                       hr=0x8007139F
                                                                                       Command-line arguments:
                                                                                       RuleId=31e71c49-8da7-4a2f-ad92-45d98a1c79ba;Action=AutoActivate;AppId=55c92734-d682-4d71-983e-d
                                                                                       6ec3f16059f;SkuId=1d02774d-66ab-4c57-8b14-e254fdce09d4;NotificationInterval=1440;Trigger=Networ
                                                                                       kQuarantineRetry
5/15/2024 10:01:24 PM  1003 Microsoft-Windows-Security-SPP                             The Software Protection service has completed licensing status check.
                                                                                       Application Id=55c92734-d682-4d71-983e-d6ec3f16059f
                                                                                       Licensing Status=
                                                                                       1: 1d02774d-66ab-4c57-8b14-e254fdce09d4, 1, 1 [(0 )(1 )(2 )(3 [0x00000000, 0, 1], [(?)( 1
                                                                                       0x00000000)( 6 0x00000000 10 14347)(?)(?)(?)( 10 0x00000000 msft:rm/algorithm/flags/1.0)(?)])]


5/15/2024 10:01:24 PM  1014 Microsoft-Windows-Security-SPP                             Acquisition of End User License failed. hr=0x80072EE7
                                                                                       Sku Id=1d02774d-66ab-4c57-8b14-e254fdce09d4
5/15/2024 10:01:24 PM  8200 Microsoft-Windows-Security-SPP                             License acquisition failure details.
                                                                                       hr=0x80072EE7
5/15/2024 10:01:24 PM  8230 Microsoft-Windows-Security-SPP                             The rules engine successfully re-evaluated the schedule.
                                                                                       Kernel policies:
                                                                                       Security-SPP-Action-StateData (REG_SZ) =AppId=55c92734-d682-4d71-983e-d6ec3f16059f;GraceEndDate
                                                                                       =2024/05/25:11:08:24;LastConsumptionReason=0x4004f00c;LastNotificationId=NeverActivated;License
                                                                                       State=SL_LICENSING_STATUS_IN_GRACE_PERIOD;PartialProductKey=37CYR;ProductKeyType=Retail:TB:Eval
                                                                                       ;SkuId=1d02774d-66ab-4c57-8b14-e254fdce09d4;ruleId=31e71c49-8da7-4a2f-ad92-45d98a1c79ba;uxDiffe
                                                                                       rentiator=TIMEBASED_EVAL

**Reply by Karl-WE** [reply]
Ah this gives me pointer . It seems that it does not like eval ISOs in the process. MS SPP (slui) is the activation process.
If you are a Developer or MS partner you might have access to Visual Studio Enterprise subscription, these have volume licensing ISOs.
Also if you are a MS customer you could access VL ISOs through admin.microsoft.com licensing plane.

**Reply by nm7s9** [reply]
Many thanks for that, I'll give VL ISO a go

**Reply by Karl-WE** [reply]
fingers crossed this is going to close this case for you and this issue.
